jsdoc/services/transforms/extract-module-name.js

```diff
@@ -0,0 +1,45 @@
+// Match [module:]moduleName [REMINDER]
+var MODULE_NAME_REGEX = /^\s*((?:module:)?[a-zA-Z0-9\._/-]+)?\s*(.*)$/;
+
+/**
+ * Extract the name information from a module tag. If such information is not present
+ * it set the module name to its file projectRelativePath without the '.js' extension.
+ * It returns what is left after matching the module name (even if nothing should remain)
+ *
+ * @param  {Tag} tag The tag to process
+ */
+module.exports = function extractModuleNameTransform(log, createDocMessage) {
+  return function(doc, tag, value) {
+    if (tag.tagName == 'module') {
+
+      // JsDoc `@module` tags does not refer to any code node
+      // TODO: implement this a generic option for any tag to be declared inside tagDefs
+      //   E.g., `{ name: 'module', noCodeNode: true, transforms: [ ... ] }`
+      doc.codeNode = null;
+      doc.codeAncestors = null;
+
+      value = value.replace(MODULE_NAME_REGEX, function(match, name, remainder) {
+        if (name) {
+          tag.name = name;
+        } else {
+          var prp = doc.fileInfo.projectRelativePath;
+          if (prp) {
+            tag.name = prp.endsWith('.js') ? prp.substring(0, prp.length - 3) : prp;
+          } else {
+            log.warn(createDocMessage("Could not set default value for '@module' tag", doc));
+          }
+        }
+
+        if (remainder) {
+          log.warn(createDocMessage(`Some text remained after parsing module name: ${remainder}`, doc));
+        }
```

Comment:
I would also not warn about this either, since it makes the transform less usable elsewhere.
If we want to ensure that that there is nothing else on the line for the `@module` tagDef then we could add another transform that did this check.